app/src/main/res/layout/fragment_main.xml

```diff
@@ -21,33 +19,69 @@
 
     </androidx.coordinatorlayout.widget.CoordinatorLayout>
 
-    <FrameLayout
-        android:id="@+id/fragment_main_nav_tab_container"
+    <LinearLayout
         android:layout_width="match_parent"
-        android:layout_height="?attr/actionBarSize"
-        android:layout_gravity="bottom"
-        android:background="?attr/nav_tab_background_color"
-        android:elevation="6dp">
+        android:layout_height="wrap_content"
+        android:orientation="horizontal"
+        android:layout_marginEnd="12dp"
+        android:baselineAligned="false">
 
-        <org.wikipedia.navtab.NavTabLayout
-            android:id="@+id/fragment_main_nav_tab_layout"
-            android:layout_width="match_parent"
-            android:layout_height="match_parent"
+        <FrameLayout
+            android:id="@+id/fragment_main_nav_tab_container"
+            android:layout_width="0dp"
+            android:layout_height="?attr/actionBarSize"
+            android:layout_gravity="bottom"
+            android:layout_weight="1"
             android:background="?attr/nav_tab_background_color"
-            app:itemIconTint="?attr/nav_tab_item_color_state"
-            app:itemTextColor="?attr/nav_tab_item_color_state"
-            app:itemTextAppearanceInactive="@style/BottomNavigationTextAppearanceInactive"
-            app:itemTextAppearanceActive="@style/BottomNavigationTextAppearanceActive"
-            app:tabGravity="fill"
-            app:labelVisibilityMode="labeled"
-            app:tabIndicatorColor="@android:color/transparent" />
-
-        <org.wikipedia.navtab.NavTabOverlayLayout
-            android:id="@+id/fragment_main_nav_tab_overlay_layout"
-            android:layout_width="match_parent"
+            android:elevation="6dp">
+
+            <org.wikipedia.navtab.NavTabLayout
+                android:id="@+id/fragment_main_nav_tab_layout"
+                android:layout_width="match_parent"
+                android:layout_height="match_parent"
+                android:background="?attr/nav_tab_background_color"
+                app:itemIconTint="?attr/nav_tab_item_color_state"
+                app:itemTextAppearanceActive="@style/BottomNavigationTextAppearanceActive"
+                app:itemTextAppearanceInactive="@style/BottomNavigationTextAppearanceInactive"
+                app:itemTextColor="?attr/nav_tab_item_color_state"
+                app:labelVisibilityMode="labeled"
+                app:tabGravity="fill"
+                app:tabIndicatorColor="@android:color/transparent" />
+
+            <org.wikipedia.navtab.NavTabOverlayLayout
+                android:id="@+id/fragment_main_nav_tab_overlay_layout"
+                android:layout_width="match_parent"
+                android:layout_height="match_parent"
+                android:elevation="8dp" />
+
+        </FrameLayout>
+
+        <LinearLayout
+            android:id="@+id/menu_layout"
+            android:layout_width="60dp"
```

Comment:
I tested it with the Samsung S9 and looks like the width is not correct because I am seeing the background color of the parent view. Also, this button should have a click ripple on this.